dataset.py

```diff
@@ -362,11 +362,21 @@ def _get_single_item(self, idx, image_height):
         image = self._load_image(idx)
 
         # Load intrinsics.
-        focal_length = float(np.loadtxt(self.calibration_files[idx]))
+        k = np.loadtxt(self.calibration_files[idx])
+        if k.size == 1:
+            focal_length = float(k)
+        else:
+            k = k.tolist()
+            focal_length = [k[0][0], k[1][1]]
+            centre_point = [k[0][2], k[1][2]]
 
         # The image will be scaled to image_height, adjust focal length as well.
         f_scale_factor = image_height / image.shape[0]
-        focal_length *= f_scale_factor
+        if centre_point:
```

Comment:
This results in a `NameError` if `centre_point` is not defined (which happens when we read a single focal length). Can you initialize it to `None` in the if/else above?